### ISSUE TITLE
[BUG] Fix False Positive "Single type parameter T should have trailing comma" 

### ISSUE THREAD

**Issue by aaronzelijah** [stylelint/stylelint#5967]
### What steps are needed to reproduce the bug?

Currently, I have a NextJS app configured with stylelint.

Here are our current versions of stylelint + any relevant config taken from the `package.json`

```
"stylelint": "13.13.1",
"stylelint-config-standard": "22.0.0",
"stylelint-config-styled-components": "0.1.1",
"stylelint-order": "4.1.0",
"stylelint-processor-styled-components": "1.10.0",
```

We configure this with a configuration file at the top-level project directory

```json
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "plugins": ["stylelint-order"],
  "processors": ["stylelint-processor-styled-components"],
  "rules": {
    "order/properties-alphabetical-order": true,
    "declaration-colon-newline-after": null,
    "no-descending-specificity": null,
    "value-keyword-case": null
  }
}
```

Running `stylelint "**/*.{ts,tsx}"` gives a listing error

```
✖  Single type parameter T should have a trailing comma. Example usage: <T,>. (23:34) 
```

The linting error shouldn't be raised by `stylelint` as this is a type parameter of a functional React component. This should be raised by `prettier` or `eslint` instead in my opinion. 

Thank you for looking into this :)

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "plugins": ["stylelint-order"],
  "processors": ["stylelint-processor-styled-components"],
  "rules": {
    "order/properties-alphabetical-order": true,
    "declaration-colon-newline-after": null,
    "no-descending-specificity": null,
    "value-keyword-case": null
  }
}
```

### How did you run Stylelint?

stylelint "**/*.{ts,tsx}"

### Which version of Stylelint are you using?

`13.13.1`

### What did you expect to happen?

I expected that no linting errors would be raised because of trailing commas (or lack thereof) on type parameters

### What actually happened?

The following linting error was reported

```
✖  Single type parameter T should have a trailing comma. Example usage: <T,>. (23:34) 
```

### Does the bug relate to non-standard syntax?

Yes, it is linting TypeScript type parameters of a React component

### Proposal to fix the bug

I saw this issue when I removed my `yarn.lock` and reinstalled and noted that certain dependencies had been bumped

`@babel/core`: `7.17.5` -> `7.17.7`.
`@babel/compat-data`: `7.17.0` -> `7.17.7`
`@babel/generator`: `7.17.3` -> `7.17.7`
`@babel/helper-compilation-targets`: `7.16.7` -> `7.17.7`
`@babel/helper-member-expression-to-functions`: `7.16.7` -> `7.17.7`
`@babel/helper-module-transforms`: `7.17.6` -> `7.17.7`
`@babel/helper-simple-access`: `7.16.7` -> `7.17.7`
`@babel/helpers`: `7.17.2` -> `7.17.7`
`@babel/parser`: `7.17.3` -> `7.17.7`
`@babel/runtime-corejs3`: `7.17.2` -> `7.17.7`
`@babel/runtime`: `7.17.2` -> `7.17.7`

**Comment by alexander-akait** [comment]
Bug in `stylelint-processor-styled-components`, `stylelint-processor-styled-components` is deprecated and should not be used
